File: PA3.java
class MyQueue {

    /** Initialize your data structure here. */
    Stack stin;
    Stack stout;
    
    public MyQueue() {
        stin = new Stack<Integer>(); 
        stout= new Stack<Integer>();
        
    }
    
    /** Push element x to the back of queue. */
    public void push(int x) {
        stin.push(x);
        
    }
    
    /** Removes the element from in front of queue and returns that element. */
    public int pop() {
        if(stout.isEmpty()){
            while(!stin.isEmpty()){
                stout.push(stin.pop());
            }
        }
        
            return (int) stout.pop();

        
    }
    
    /** Get the front element. */
    public int peek() {
        if(stout.isEmpty()){
            while(!stin.isEmpty()){
                stout.push(stin.pop());
            }
        }
        
            return (int)stout.peek();
    }
    
    /** Returns whether the queue is empty. */
    public boolean empty() {
        return stin.isEmpty() && stout.isEmpty();
    }
}

/**
 * Your MyQueue object will be instantiated and called as such:
 * MyQueue obj = new MyQueue();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.peek();
 * boolean param_4 = obj.empty();
 */

File: PA_3.py
#Leetcode 232 : Implement Queue using Stacks
#Space Complexity - O(2n) [Overall]
class MyQueue(object):

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.stack1 = []
        self.stack2 = []
        

    def push(self, x):
        """
        Push element x to the back of queue.
        :type x: int
        :rtype: None
        """
        #Time Complexity - O(1)
        self.stack1.append(x)

    def pop(self):
        """
        Removes the element from in front of queue and returns that element.
        :rtype: int
        """
        #Time Complexity - O(1) if Stack2 is not empty, else O(n)
        self.peek()
        x = self.stack2.pop()
        return x

    def peek(self):
        """
        Get the front element.
        :rtype: int
        """
        #Time Complexity - O(1) if Stack2 is not empty, else O(n)
        if self.stack2 == []:
            while self.stack1:
                self.stack2.append(self.stack1.pop())
        return self.stack2[-1]

    def empty(self):
        """
        Returns whether the queue is empty.
        :rtype: bool
        """
        #Time Complexity - O(1)
        return self.stack1 == [] and self.stack2 == []